components/brave_rewards/browser/test/rewards_p3a_browsertest.cc

```diff
@@ -140,6 +175,86 @@ IN_PROC_BROWSER_TEST_F(RewardsP3ABrowserTest, RewardsDisabled) {
   histogram_tester_->ExpectBucketCount("Brave.Rewards.AutoContributionsState.2",
                                        1, 1);
   histogram_tester_->ExpectBucketCount("Brave.Rewards.TipsState.2", 1, 1);
+  histogram_tester_->ExpectBucketCount("Brave.Rewards.EnabledDuration",
+                                       RewardsEnabledDuration::kNever, 1);
+}
+
+IN_PROC_BROWSER_TEST_F(RewardsP3ABrowserTest, Duration) {
+  rewards_browsertest_util::StartProcess(rewards_service_);
+  WaitForRewardsInitialization();
+
+  PrefService* prefs = browser()->profile()->GetPrefs();
+
+  // Turn rewards on.
+  rewards_service_->SetAdsEnabled(true);
+  WaitForRewardsEnable();
+  histogram_tester_->ExpectBucketCount("Brave.Rewards.EnabledDuration",
+                                       RewardsEnabledDuration::kStillEnabled,
+                                       1);
+
+  // We can't turn rewards back off without shutting down the ledger
+  // process, which interferes with other tests running in parallel.
+  // Instead rely on the fact that the EnabledDuration P3A measurement
+  // is made by the rewards service preference observer.
+  prefs->SetBoolean("brave.brave_ads.enabled", false);
+  histogram_tester_->ExpectBucketCount("Brave.Rewards.EnabledDuration",
+                                       RewardsEnabledDuration::kHours, 1);
+
+  // Mock turning rewards back on.
+  prefs->SetBoolean("brave.brave_ads.enabled", true);
+  // Adjust the stored timestamp to measure a longer duration.
+  auto earlier = base::Time::Now() - base::Minutes(90);
+  LOG(INFO) << "Backdating timestamp to " << earlier;
```

Comment:
VLOG ?

Reply:
I think `LOG` is better for tests.  The output is normally captured by the test harness, so it doesn't clutter normal runs, but is then shown as part of the report when a test fails, without having to figure out how to invoke the test with `-v=1` or whatever.

Nevertheless, replaced in  9d97e1cc3ddc2a343040036205bef9f372355cc7.